File: 260_single_number_3/singleNumber.py
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        hash = {}

        res = []

        for i in range(len(nums)):
            hash[nums[i]] = hash.get(nums[i], 0) + 1

        for key, item in hash.items():
            if item == 1:
                res.append(key)
        return res

File: 260_single_number_3/singleNumber.java
class Solution {
    public int[] singleNumber(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();

        int[] result = new int[] { 0, 0 };

        for (int i = 0; i < nums.length; i++) {
            int value = map.getOrDefault(nums[i], 0);
            map.put(nums[i], ++value);
        }
        int idx = 0;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            Integer key = entry.getKey();
            Integer value = entry.getValue();
            if (value == 1 && idx < 2) {
                result[idx] = key;
                idx++;
            }
        }
        return result;
    }
}